### PR TITLE
Add reconfiguration action handling for unchanged and added tasks

### PR DESCRIPTION
**Pull request by mklkun**

### CODE REVIEW

File: sal-service/src/main/java/org/ow2/proactive/sal/service/service/MonitoringService.java [hunk 1]
```diff
@@ -53,12 +57,19 @@ public class MonitoringService {
     @Autowired
     private RepositoryService repositoryService;
 
+    private String baguetteIp;
+
+    private static final int BAGUETTE_PORT = 8111;
```
Comment (fviale) [inline sal-service/src/main/java/org/ow2/proactive/sal/service/service/MonitoringService.java:62]: baguette port ?

File: sal-service/src/main/java/org/ow2/proactive/sal/service/service/NodeService.java [hunk 2]
```diff
@@ -72,109 +73,119 @@ public class NodeService {
...
+        if (cloud == null) {
+            LOGGER.error("Cloud [{}] does not exist.", node.getCloudId());
+            throw new IllegalArgumentException(String.format("node.cloudId [%s] not valid.", node.getCloudId()));
+        }
+        cloud.addDeployment(newDeployment);
+        if (WhiteListedInstanceTypesUtils.isHandledHardwareInstanceType(newDeployment.getNode()
+                                                                                     .getNodeCandidate()
+                                                                                     .getHardware()
+                                                                                     .getProviderId())) {
+            if (Boolean.FALSE.equals(cloud.isWhiteListedRegionDeployed(newDeployment.getNode()
```
Comment (fviale) [inline sal-service/src/main/java/org/ow2/proactive/sal/service/service/NodeService.java:136]: can't you use autoboxing ?
e.g.
```
if (!cloud.isWhiteListedRegionDeployed(newDeployment.getNode()
                                                                                    .getNodeCandidate()
                                                                                    .getLocation()
                                                                                    .getName()))
```

Comment (mklkun) [reply sal-service/src/main/java/org/ow2/proactive/sal/service/service/NodeService.java:136]: This is to shut-up the sonar smell regarding booleans testing condition.
See [here](https://stackoverflow.com/questions/59035202/sonarlint-use-the-primitive-boolean-expression-here)

File: sal-service/src/main/java/org/ow2/proactive/sal/service/service/MonitoringService.java (hunk 1 above)
Comment (mklkun) [reply sal-service/src/main/java/org/ow2/proactive/sal/service/service/MonitoringService.java:62]: lool This is the terminology used by EMS monitoring component (implemented by partners).

**Review by alijawadfahs — APPROVED** [review]
LGTM

File: sal-service/src/main/java/org/ow2/proactive/sal/service/service/NodeService.java (hunk 2 above)
Comment (fviale) [reply sal-service/src/main/java/org/ow2/proactive/sal/service/service/NodeService.java:136]: ok if there can be null ...

**Review by fviale — APPROVED** [review]

**Pull request merged**